### PR TITLE
feat(rome_cli): Add a new option --config-path to set the path to rome.json

### PR DESCRIPTION
**Pull request by realtimetodie**
Adds a new option --config-path to set the filesystem path to the directory of the rome.json configuration file.

Example

```bash
$ rome check --config-path=../../../example
```

The --config option appends a base path to the rome.json configuration file path

```
../../../example/rome.json
```

Error handling

If the user explicitly sets a path to the rome.json configuration file using the --config-path option and the file is not found, an error will be thrown.

Example

```bash
$ rome check --config-path=/test.json
```

Error message

```
...
× Rome couldn't read the following file, maybe for permissions reasons or it doesn't exists: /test.json/rome.json
```

- [x] The PR requires documentation
- [x] I will create a new PR to update the documentation

### CODE REVIEW

**Comment by realtimetodie** [comment]
This is required for the Bazel [rules_rome](https://github.com/realtimetodie/rules_rome) in order to run the Rome cli as a Bazel tool.

Rome can be run under Bazel to edit the source files in place instead of printing a result.

Example

```bash
$ bazel run @build_bazel_rules_rome//rome:check
```

This is equivalent to running the Rome linter with the --write option

```bash
$ rome check --write
```

**Comment by realtimetodie** [comment]
This is required for the Bazel [rules_rome](https://github.com/realtimetodie/rules_rome) in order to run the Rome cli as a Bazel tool.

Rome can be run under Bazel to edit the source files in place instead of printing a result.

Example

```bash
$ bazel run @build_bazel_rules_rome//tools:check
```

This is equivalent to running the Rome linter with the --write option

```bash
$ rome check --write
```

**Comment by realtimetodie** [comment]
I had to flip `ConfigurationStatus::Error` to `ConfigurationStatus::Missing`, as the base path is always set in the lsp server.

**Comment by realtimetodie** [comment]
This is required for the Bazel [rules_rome](https://github.com/realtimetodie/rules_rome) in order to run the Rome cli as a Bazel tool.

Rome can be run under Bazel to edit the source files in place instead of printing a result.

Example

```bash
$ bazel run @build_bazel_rules_rome//tools:check
```

This is equivalent to running the Rome linter with the --write option

```bash
$ rome check --write
```

**Comment by realtimetodie** [comment]
@MichaReiser @leops @ematipico Please let me know how to continue, before this pull request gets marked as stalled. This is blocking me. I would like to continue to work on the video about Rome + Bazel for my upcoming talk.

**Review by ematipico — CHANGES_REQUESTED** [review]
Thank you for this PR! I think we should rename the option, for example `--config-path`. This would be in line with `--file-path`

**Comment by realtimetodie** [comment]
@ematipico Yes, I like this idea. I renamed the option to --config-path and updated the documentation.

File: crates/rome_cli/src/configuration.rs [hunk 1]
```diff
@@ -7,5 +9,12 @@ use rome_service::{load_config, Configuration};
...
 ) -> Result<Deserialized<Configuration>, CliDiagnostic> {
-    Ok(load_config(&session.app.fs, None)?.unwrap_or_default())
+    let config_path: Option<PathBuf> = session
+        .args
+        .opt_value_from_str("--config-path")
+        .map_err(|source| CliDiagnostic::parse_error("--config-path", source))?;
+
+    let is_config_path = config_path.is_some();
+
+    Ok(load_config(&session.app.fs, config_path, is_config_path)?.unwrap_or_default())
```
Comment (ematipico) [inline crates/rome_cli/src/configuration.rs:19]: Passing a flag just to show an error doesn't seem right to me. Maybe, we should revisit a bit the `load_config` function a bit and send a data structure that holds more information:

```rust
pub fn load_config(file_system: &DynRef<dyn FileSystem>, data: LoadData) -> LoadConfig {}


/// Metadata data of how to load a configuration file
pub struct LoadData {
    /// custom base path where to load `rome.json` file
    base_path: Option<PathBuf>,
    /// how to load the configuration
    load_kind: LoadKind,
}

impl From<PathBuf> for LoadData {
    fn from(base_path: PathBuf) -> Self {
        Self {
            base_path: Some(base_path),
            load_kind: LoadKind::default(),
        }
    }
}

#[derive(Default)]
enum LoadKind {
    /// Default mode, it throws errors only when the file is not found
    #[default]
    Default,
    /// Base path of the LSP, it should ignore errors
    Lsp,
    /// Path provided from the user, if should fail at any kind of error
    FromUser,
}

```
And then, we can match `LoadKind` and log based on its variants. For example, the case of `LoadKind::FromUser` should always throw a diagnostic for any error. While the solution provided does the same, my suggestion allows for better documentation and maintainability in the long run.

**Review by ematipico — COMMENTED** [review]
Have you tried running the CLI using windows like style paths? E.g. `--config-path=..\foo\rome.json`

Does it work?

File: crates/rome_cli/src/configuration.rs (hunk 1 above)
Comment (realtimetodie) [reply crates/rome_cli/src/configuration.rs:19]: This turns the binary logic into a complex three state logic.

**Comment by realtimetodie** [comment]
> Have you tried running the CLI using windows like style paths? E.g. `--config-path=..\foo\rome.json`

The dot notation is covered by the DOS specification

```rust
assert_eq!(Path::new("..\\..").join("rome.json"), PathBuf::from("..\\..\\rome.json"));
```

File: crates/rome_cli/src/configuration.rs (hunk 1 above)
Comment (ematipico) [reply crates/rome_cli/src/configuration.rs:19]: Yeprecisely's exactly what _I think_ how it should be. But if you think it's too much logic, then I'd suggest renaming `show_error` in `is_user_custom_path` or something similar. The `show_error` name doesn't provide enough context of **why** we should show an error. Plus, could you please update the documentation of the function? The logic is now slightly different, and we should document that.

**Review by ematipico — APPROVED** [review]

**Comment by ematipico** [comment]
@realtimetodie could you please tick the checkbox of `The PR requires documentation`? We would need to update the website with this new argument

**Comment by realtimetodie** [comment]
@ematipico Yes, I will take care about the documentation.

File: crates/rome_cli/src/configuration.rs (hunk 1 above)
Comment (realtimetodie) [reply crates/rome_cli/src/configuration.rs]: I took your suggestion at heart and came up with this. We can store the information in a simple enum data type.

```rust
pub enum BasePath {
    #[default]
    None,
    Lsp(PathBuf),
    FromUser(PathBuf),
}
```

```rust
pub fn load_config(file_system: &DynRef<dyn FileSystem>, base_path: BasePath) -> LoadConfig { ... }
```

Let me know what you think.

**Comment by realtimetodie** [comment]
@ematipico https://github.com/rome/tools/pull/4158#discussion_r1104836487

File: crates/rome_cli/src/configuration.rs (hunk 1 above)
Comment (ematipico) [reply crates/rome_cli/src/configuration.rs]: That's way better! Great way to take a suggestion and improve it!

**Pull request merged**